pkg/tikvapi/tikvapi.go

```diff
@@ -36,12 +43,69 @@ const (
 // TiKVClient provides tikv server's api
 type TiKVClient interface {
 	GetLeaderCount() (int, error)
+	FlushLogBackupTasks(ctx context.Context) error
+}
+
+type lazyGRPCConn struct {
+	target string
+	opts   []grpc.DialOption
+}
+
+func (l *lazyGRPCConn) conn(ctx context.Context) (*grpc.ClientConn, error) {
+	ctx, cancel := context.WithTimeout(ctx, DefaultTimeout)
+	defer cancel()
+	ch, err := grpc.DialContext(ctx, l.target, l.opts...)
+	if err != nil {
+		return nil, errors.Annotatef(err, "during connecting to %s", l.target)
+	}
+
+	return ch, nil
 }
 
 // tikvClient is default implementation of TiKVClient
 type tikvClient struct {
-	url        string
-	httpClient *http.Client
+	url           string
+	httpClient    *http.Client
+	grpcConnector *lazyGRPCConn
+}
+
+// FlushLogBackupTasks implements TiKVClient.
+func (c *tikvClient) FlushLogBackupTasks(ctx context.Context) error {
+	logger := klog.FromContext(ctx)
+
+	// For now we are using one-shot sessions, because the `TiKVClient`
+	// interface doesn't provide a `Close` method...
+	conn, err := c.grpcConnector.conn(ctx)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := conn.Close(); err != nil {
+			logger.Error(err, "tikvClient: failed to close grpc connection")
+		}
+	}()
+
+	cli := logbackup.NewLogBackupClient(conn)
+	res, err := cli.FlushNow(ctx, &logbackup.FlushNowRequest{})
+	if err != nil {
+		return err
+	}
+
+	// Fast path: no task, return early.
+	if len(res.Results) == 0 {
+		return nil
+	}
+
+	for _, r := range res.Results {
+		if r.Success {
+			logger.Info("successfully flushed the log backup task.", "task", r.TaskName)
+		} else {
+			return errors.Errorf("force flush failed for task %s: %s", r.TaskName, r.ErrorMessage)
+		}
+	}
+	// Wait a while so TiKV is able to send the flush result to the advancer.
+	time.Sleep(3 * time.Second)
```

Comment:
may i know how "3 seconds" is determined?

Reply:
Just a magic number...

Reply:
how about define a constant for it for better readability?

Reply:
Can we return a triable error and let operator requeue and retry?

Reply:
It is somehow tricky to make it requeue because I think we need some sort of extra status to make sure that it was flushed. Perhaps this sleep can be finished in TiKV side...